nixos/modules/services/networking/doh-proxy-rust.nix

```diff
@@ -0,0 +1,69 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.doh-proxy-rust;
+
+in {
+
+  options.services.doh-proxy-rust = {
+
+    enable = mkEnableOption "doh-proxy-rust";
+
+    flags = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = literalExample [ "--server-address=9.9.9.9:53" ];
+      description = ''
+        A list of command-line flags to pass to doh-proxy.
+      '';
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    users.users.doh-proxy-rust = {
+      description = "doh-proxy-rust service user";
+    };
+
+    systemd.services.doh-proxy-rust = {
+      description = "doh-proxy-rust";
+      after = [ "network.target" "nss-lookup.target" ];
+      wantedBy = [ "multi-user.target" ];
+      confinement = {
```

Comment:
While I'm sure @aszlig would love to see this... we don't actually use `confinement` anywhere else in our module system.

@flokli @andir @arianvp @Mic92 what do you all think about using `confinement` in upstream NixOS modules?